blockly/generators/arduino/control.js

```diff
@@ -75,3 +75,20 @@ Blockly.Arduino.controls_for = function() {
   }
   return code;
 };
+
+Blockly.Arduino.controls_whileUntil = function(){
```

Comment:
Nope! Surprisingly the block definition was already made for this. Take a peek inside of blocks/loop.js. It should be the third one.